src/codeu/chat/client/simplegui/ConversationPanel.java

```diff
@@ -134,6 +137,26 @@ public void actionPerformed(ActionEvent e) {
       }
     });
 
+    // User clicks Conversations Rename button.
+    renameButton.addActionListener(new ActionListener() {
+      @Override
+      public void actionPerformed(ActionEvent e) {
+        if(objectList.getSelectedIndex() != -1) {
+          final int index = objectList.getSelectedIndex();
+          final String data = objectList.getSelectedValue();
+          final ConversationSummary cs = ConversationPanel.this.lookupByTitle(data, index);
+
+          final String title = JOptionPane.showInputDialog(ConversationPanel.this, "Enter new title:",
+            "Rename Conversation", JOptionPane.PLAIN_MESSAGE);
+
+          clientContext.conversation.renameConversation(title);
```

Comment:
Inside this conversation that you are renaming, do you also want to post a message saying [user] has renamed the conversation to [newName]?

Reply:
Yea, that makes sense, added.